contrib/rtlbrowse/logfile.c

```diff
@@ -93,1390 +89,1308 @@ static GtkTextTag *blue_tag = NULL, *fwht_tag = NULL;
-    g_signal_connect(find_entry, "key_press_event", (GCallback) find_edit_cb, NULL);
-    gtk_box_pack_start (GTK_BOX (hbox), find_entry, FALSE, FALSE, 0);
+    find_label = gtk_label_new("Find:");
+    gtk_box_append(GTK_BOX(hbox), find_label);
 
-    tb = gtk_toolbar_new();
-    gtk_widget_show (tb);
+    find_entry = gtk_search_entry_new();
+    g_signal_connect(find_entry, "search_changed", G_CALLBACK(find_edit_cb), (void*)TRUE);
+    g_signal_connect (find_entry, "activate", G_CALLBACK (find_edit_cb), FALSE);
 
-    gtk_toolbar_set_style(GTK_TOOLBAR(tb), GTK_TOOLBAR_ICONS);
-    stock = XXX_gtk_toolbar_insert_stock(GTK_TOOLBAR(tb),
-                                                 XXX_GTK_STOCK_GO_BACK,
-                                                 "Search Back",
-                                                 NULL,
-                                                 G_CALLBACK(search_backward),
-                                                 NULL,
-                                                 tb_pos++);
+    gtk_box_append(GTK_BOX(hbox), find_entry);
 
-    gtk_widget_show(stock);
+    box = gtk_box_new(GTK_ORIENTATION_HORIZONTAL, 8);
 
-    stock = XXX_gtk_toolbar_insert_stock(GTK_TOOLBAR(tb),
-                                                 XXX_GTK_STOCK_GO_FORWARD,
-                                                 "Search Forward",
-                                                 NULL,
-                                                 G_CALLBACK(search_forward),
-                                                 NULL,
-                                                 tb_pos); /* scan-build: removed ++ on tb_pos as never read */
+    create_button_in_box(GTK_WIDGET(box),
+                         XXX_GTK_STOCK_GO_BACK,
+                         "Search Back",
+                         G_CALLBACK(search_backward),
+                         NULL);
 
-    gtk_widget_show(stock);
+    create_button_in_box(GTK_WIDGET(box),
+                         XXX_GTK_STOCK_GO_FORWARD,
+                         "Search Forward",
+                         G_CALLBACK(search_forward),
+                         NULL);
 
-    gtk_box_pack_start (GTK_BOX (hbox), tb, FALSE, FALSE, 0);
+    gtk_box_append(GTK_BOX(hbox), box);
 
     matchcase_checkbutton = gtk_check_button_new_with_label("Match case");
-    gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(matchcase_checkbutton), matchcase_active);
-    g_signal_connect (matchcase_checkbutton, "toggled", G_CALLBACK(toggle_callback), NULL);
-    gtk_widget_show(matchcase_checkbutton);
-    gtk_box_pack_start (GTK_BOX (hbox), matchcase_checkbutton, FALSE, FALSE, 0);
-    }
-
+    gtk_check_button_set_active(GTK_CHECK_BUTTON(matchcase_checkbutton), matchcase_active);
+    g_signal_connect(matchcase_checkbutton, "toggled", G_CALLBACK(toggle_callback), NULL);
 
+    gtk_box_append(GTK_BOX(hbox), matchcase_checkbutton);
+}
 
 static char *tmpnam_rtlbrowse(char *s, int *fd)
```

Comment:
`tmpnam_rltbrowse` was used by drag and drop, but I don't think we'll use it in the future and this can be removed.

Reply:
solved in f70a7043